### PR TITLE
Added a setup.py, and README.rst to work with PyPI

### PR DESCRIPTION
**Pull request by timcera**
Added a setup.py and moved README.md to README.rst and other misc. so that it will work with PyPI, pip, 'python setup.py install', ...etc.  Needs testing since I didn't want to load up to PyPI leaving that decision to you guys.  The 'pip' and 'python setup.py install' methods work on my Linux development machine.  Creates a 'hsp2' executable which should install correctly on Linux, Windows (where it makes a 'hsp2.exe') and Mac.  I created a 'run' sub-command so:

    hsp2 run tutorial.h5

runs the model....

### CODE REVIEW

File: HSP2/main.py [hunk 1]
```diff
@@ -14,19 +14,35 @@
...
 import os
 import importlib
+from datetime import datetime as dt
+from collections import defaultdict
+
 import h5py
 import pandas as pd
 import numpy as np
-from datetime import datetime as dt
-from collections import defaultdict
```
Comment (rheaphy) [inline HSP2/main.py]: Not sure why these two lines moved? Is there any significance?

File: HSP2/main.py [hunk 2]
```diff
@@ -14,19 +14,35 @@
...
+        saveall=False,
+        reloadkeys=False):
+    '''Runs main HSP2 program.
+
+    Parameters
+    ----------
+    hdfname: str
+        HDF5 filename for both input and
+        output.
+    saveall
```
Comment (rheaphy) [inline HSP2/main.py]: The original line for saveall indicates this is an optional argument.  I do not expect users to change this in general.  Having a default argument does not automatically carry this information.  I would prefer an indication this is a optional.

File: HSP2/main.py [hunk 3]
```diff
@@ -14,19 +14,35 @@
...
+        saves all calculated data ignoring SAVE
+        tables
+    reloadkeys
```
Comment (rheaphy) [inline HSP2/main.py]: Also an optional argument as above

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
+
```
Comment (rheaphy) [inline setup.py]: Thank you for this!

File: HSP2/main.py (hunk 1 above)
Comment (timcera) [reply HSP2/main.py]: Nothing more mysterious than that I group Python imports together.

File: HSP2/main.py (hunk 3 above)
Comment (timcera) [reply HSP2/main.py]: Easy fix - will do.

File: HSP2/main.py (hunk 2 above)
Comment (timcera) [reply HSP2/main.py]: Easy fix - will do.

**Comment by timcera** [comment]
Bump.

**Pull request merged**